app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
+  belongs_to :activity, inverse_of: :notifications, dependent: :destroy
+
+  acts_as_readable on: :updated_at
+
+  def self.notify(user, course, object, types = {})
```

Comment:
Document your API calls. I won't be reading your PR to see how to use this method.